AirCasting/SessionViews/ExternalSessionsViews/ExternalSessionCard.swift

```diff
@@ -26,43 +26,51 @@ struct ExternalSessionCard: View {
     @State private var isCollapsed: Bool
     @State private var selectedStream: MeasurementStreamEntity?
     @State private var isMapButtonActive = false
+    @State private var isGraphButtonActive = false
     @EnvironmentObject var reorderButton: ReorderButton
     @EnvironmentObject var searchAndFollowButton: SearchAndFollowButton
-
+    
     @StateObject private var mapStatsDataSource: ConveringStatisticsDataSourceDecorator<MapStatsDataSource>
     @StateObject private var mapStatsViewModel: StatisticsContainerViewModel
-
+    @StateObject private var graphStatsDataSource: ConveringStatisticsDataSourceDecorator<GraphStatsDataSource>
+    @StateObject private var graphStatsViewModel: StatisticsContainerViewModel
+    
     @ObservedObject var viewModel = ExternalSessionCardViewModel()
-
+    
     @Injected private var uiStateHandler: SessionCardUIStateHandler
-
+    
     init(session: ExternalSessionEntity, thresholds: [SensorThreshold]) {
         self.session = session
         self.thresholds = thresholds
         self._isCollapsed = .init(initialValue: !(session.uiState?.expandedCard ?? false))
         let mapDataSource = ConveringStatisticsDataSourceDecorator<MapStatsDataSource>(dataSource: MapStatsDataSource(), stream: nil)
         self._mapStatsDataSource = .init(wrappedValue: mapDataSource)
         self._mapStatsViewModel = .init(wrappedValue: ExternalSessionCard.createStatsContainerViewModel(dataSource: mapDataSource))
+        let graphDataSource = ConveringStatisticsDataSourceDecorator<GraphStatsDataSource>(dataSource: GraphStatsDataSource(), stream: nil)
+        self._graphStatsDataSource = .init(wrappedValue: graphDataSource)
+        self._graphStatsViewModel = .init(wrappedValue: ExternalSessionCard.createStatsContainerViewModel(dataSource: graphDataSource))
     }
-
+    
     var streams: [MeasurementStreamEntity] {
         session.sortedStreams
     }
-
+    
     var body: some View {
         sessionCard
             .onAppear(perform: { selectDefaultStreamIfNeeded(streams: session.allStreams) })
             .onChange(of: selectedStream, perform: { [weak mapStatsDataSource] newStream in
                 mapStatsDataSource?.stream = newStream
                 mapStatsDataSource?.dataSource.stream = newStream
                 uiStateHandler.changeSelectedStream(sessionUUID: session.uuid, newStream: newStream?.sensorName ?? "")
+                graphStatsDataSource.stream = newStream
+                graphStatsDataSource.dataSource.stream = newStream
             })
             .onChange(of: isMapButtonActive) { _ in
                 reorderButton.setHidden(if: isMapButtonActive)
                 searchAndFollowButton.setHidden(if: isMapButtonActive)
             }
     }
```

Comment:
You can also add here the below code to resolve bugs with appearing icons when the user will swipe from the graph.

`.onChange(of: isGraphButtonActive) { _ in
            reorderButton.setHidden(if: isGraphButtonActive)
            searchAndFollowButton.setHidden(if: isGraphButtonActive)
        }`

Reply:
Thank you 💙